packages/calling-stateful-client/src/BetaToStableTypes.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// This is for mapping new types from beta, which does not exist in stable sdk.
+// These mappings are necessary to bypass type check of conditional compilation
+// Remove these types when the feature get stabilized in stable sdk
+
+// For example, CallCommon doesn't exist in stable call sdk, so it will be still Call type
+// So we remove CallCommonBeta type in stable
+// In beta, Call | CallCommon = CallCommon because Call is just a super set of CallCommon
+
+import { Call, CallAgent, IncomingCall } from '@azure/communication-calling';
+/* @conditional-compile-remove(teams-identity-support) */
+import {
+  TeamsCall as TeamsCallBeta,
+  CallCommon as CallCommonBeta,
+  CallAgentCommon as CallAgentCommonBeta,
+  TeamsCallAgent as TeamsCallAgentBeta,
+  IncomingCallCommon as IncomingCallCommonBeta
+} from '@azure/communication-calling';
+
+/**
+ * @private
+ */
+export type CallCommon = Call | /* @conditional-compile-remove(teams-identity-support) */ CallCommonBeta;
+
+/**
+ * @private
+ */
+export type CallAgentCommon = CallAgent | /* @conditional-compile-remove(teams-identity-support) */ CallAgentCommonBeta;
+
+/**
+ * @private
+ */
+export type TeamsCall = never | /* @conditional-compile-remove(teams-identity-support) */ TeamsCallBeta;
```

Comment:
Can't we just conditionally compile this export since this won't be imported from BetaToStableStypes.ts in stable

Reply:
Is it because we can't conditionally compile the file TeamsCallAgentDeclarative.ts where TeamsCallAgent is imported?

Reply:
If we do that for the whole file, we will have to conditional compile every line of that file